core/input/input.cpp

```diff
@@ -1299,19 +1299,19 @@ void Input::parse_mapping(String p_mapping) {
 		switch (input[0]) {
 			case 'b':
 				binding.inputType = TYPE_BUTTON;
-				binding.input.button = input.substr(1).to_int();
+				binding.input.button = (JoyButton)input.substr(1).to_int();
 				break;
 			case 'a':
 				binding.inputType = TYPE_AXIS;
-				binding.input.axis.axis = input.substr(1).to_int();
+				binding.input.axis.axis = (JoyAxis)input.substr(1).to_int();
 				binding.input.axis.range = input_range;
 				binding.input.axis.invert = invert_axis;
 				break;
 			case 'h':
 				ERR_CONTINUE_MSG(input.length() != 4 || input[2] != '.',
 						String(entry[idx] + "\nInvalid hat input: " + input));
 				binding.inputType = TYPE_HAT;
-				binding.input.hat.hat = input.substr(1, 1).to_int();
+				binding.input.hat.hat = (HatDir)input.substr(1, 1).to_int();
 				binding.input.hat.hat_mask = static_cast<HatMask>(input.substr(3).to_int());
```

Comment:
Shouldn't `static_cast<>` be used everywhere instead of C-like casting? :thinking:

Reply:
I'm not sure which is desired, it's up to @akien-mga's personal preference I guess.

Reply:
From what I seen, C-like casting is preferred in the codebase. I don't remember seeing `static_cast` aside from the line above.

Reply:
We tend to use C-style casting yes, though I'm not sure why specifically (mostly habit/preference for readability).